.github/workflows/superset-backend.yml

```diff
@@ -0,0 +1,151 @@
+name: Backend
+
+on:
+  # only build on direct push to `master` branch
+  push:
+    branches: [ master ]
+    paths:
+    - superset/**/*.py
+    - tests/**
+    - setup.py
+    - requirements*.txt
+  # but also build on pull requests to any branch
+  # (the so-called feature branch)
+  pull_request:
+    paths:
+    - superset/**/*.py
+    - tests/**
+    - setup.py
+    - requirements*.txt
+
+jobs:
+  python-lint:
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        python-version: [3.6]
+    env:
+      PYTHON_LINT_TARGET: setup.py superset tests
+      CI: github-actions
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Cache mypy
+      uses: actions/cache@v1
+      with:
+        path: .mypy_cache
+        key: ${{ runner.os }}-mypy-${{ hashFiles('./requirements*.txt') }}
+        restore-keys: |
+          ${{ runner.os }}-mypy-
+    - name: Setup Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      uses: ktmud/cached-dependencies@v1
```

Comment:
can we name this something like `superset/cached-dependencies@v1`?

Reply:
Also, make it an entity reference as to DRY things up. If you need to bust the cache later, this should simplify that.

Reply:
Can you clarify on `entity reference`? I think I've optimized the cache setup as best as I can.

Reply:
See where these references are defined and then used later down the file:

https://github.com/apache/incubator-superset/blob/master/docker-compose.yml#L26

Reply:
Oh, so this is a feature in YAML. Didn't know you can do that. Thanks!

Reply:
Just spend an hour refactor an got this: 

```
### ERRORED 21:06:58Z

- Your workflow file was invalid: The workflow is not valid. .github/workflows/superset-python.yml: Anchors are not currently supported. Remove the anchor 'python-job'
```

😞 Bummer!

Reply:
Oof